subgraphs/hop-protocol/protocols/hop-protocol/src/L2/bridge.ts

```diff
@@ -0,0 +1,268 @@
+import { SDK } from "../../../../src/sdk/protocols/bridge";
+import { TokenPricer } from "../../../../src/sdk/protocols/config";
+import {
+  TokenInitializer,
+  TokenParams,
+} from "../../../../src/sdk/protocols/bridge/tokens";
+import {
+  BridgePermissionType,
+  BridgePoolType,
+  CrosschainTokenType,
+} from "../../../../src/sdk/protocols/bridge/enums";
+import { reverseChainIDs } from "../../../../src/sdk/protocols/bridge/chainIds";
+import { BridgeConfig } from "../../../../src/sdk/protocols/bridge/config";
+import { Versions } from "../../../../src/versions";
+import { NetworkConfigs } from "../../../../configurations/configure";
+import {
+  Address,
+  BigDecimal,
+  BigInt,
+  Bytes,
+  dataSource,
+  log,
+} from "@graphprotocol/graph-ts";
+import {
+  TransferSent,
+  TransferFromL1Completed,
+} from "../../../../generated/Bridge/Bridge";
+import { Token } from "../../../../generated/schema";
+import { getUsdPricePerToken, getUsdPrice } from "../../../../src/prices/index";
+import { bigIntToBigDecimal } from "../../../../src/sdk/util/numbers";
+
+import { Network } from "../../../../src/sdk/util/constants";
+import {
+  updateL2OutgoingBridgeMessage,
+  updateL2IncomingBridgeMessage,
+} from "../../../../src/sdk/util/bridge";
+
+class Pricer implements TokenPricer {
+  getTokenPrice(token: Token): BigDecimal {
+    const price = getUsdPricePerToken(Address.fromBytes(token.id));
+    return price.usdPrice;
+  }
+
+  getAmountValueUSD(token: Token, amount: BigInt): BigDecimal {
+    const _amount = bigIntToBigDecimal(amount, token.decimals);
+    return getUsdPrice(Address.fromBytes(token.id), _amount);
+  }
+}
+
+class TokenInit implements TokenInitializer {
+  getTokenParams(address: Address): TokenParams {
+    const tokenConfig = NetworkConfigs.getTokenDetails(address.toHex());
+    const name = tokenConfig[1];
+    const symbol = tokenConfig[0];
+    const decimals = BigInt.fromString(tokenConfig[2]).toI32();
+    return { name, symbol, decimals };
+  }
+}
+const conf = new BridgeConfig(
+  "0x03d7f750777ec48d39d080b020d83eb2cb4e3547",
+  "HOP-"
+    .concat(dataSource.network().toUpperCase().replace("-", "_"))
+    .concat("-BRIDGE"),
+  "hop-".concat(dataSource.network().replace("-", "_")).concat("-bridge"),
+  BridgePermissionType.PERMISSIONLESS,
+  Versions
+);
+
+export function handleTransferFromL1Completed(
+  event: TransferFromL1Completed
+): void {
+  log.warning("TransferFromL1 - bridgeAddress: {},  hash: {}", [
```

Comment:
Most of the warnings should probably just be info if you still want to log them: https://github.com/messari/subgraphs/blob/master/docs/WALKTHROUGH.md#using-logs-in-subgraphs

Reply:
Okay, I'll change them